7.External-Audio-Device/External-Audio-Device/OTDefaultAudioDevice.m

```diff
@@ -1094,6 +1085,20 @@ - (BOOL)configureAudioSessionWithDesiredAudioRoute:(NSString *)desiredAudioRoute
                                         error:&err];
     }
     
+    if([AUDIO_DEVICE_HEADSET isEqualToString: desiredAudioRoute]) {
```

Comment:
See my comment on the pull request.
This was tested on iPhone 5 iOS 7.
Meaning these blocked the code from executing and it did not work.

So it's my bad. The guards and the re-ordering need to be there.

On Mon, Jun 15, 2015 at 9:13 PM, Charley Robinson <notifications@github.com>
wrote:

> In 7.External-Audio-Device/External-Audio-Device/OTDefaultAudioDevice.m
> <https://github.com/opentok/opentok-ios-sdk-samples/pull/97#discussion_r32449490>
> :
>
> > @@ -1094,6 +1085,20 @@ - (BOOL)configureAudioSessionWithDesiredAudioRoute:(NSString *)desiredAudioRoute
> >                                          error:&err];
> >      }
> >
> > +    if([AUDIO_DEVICE_HEADSET isEqualToString: desiredAudioRoute]) {
>
> This seems like the only actual change other than the reordering of the
> setActive: calls. Re-add the removed iOS7 guards around setActive:, you
> can keep the reordering, otherwise lgtm.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opentok/opentok-ios-sdk-samples/pull/97/files#r32449490>
> .
>